workflows/sarek/main.nf

```diff
@@ -149,10 +154,34 @@ workflow SAREK {
 
         // Figure out if input is bam or fastq
         input_sample_type = input_sample.branch{
-            bam:   it[0].data_type == "bam"
-            fastq: it[0].data_type == "fastq"
+            bam:                 it[0].data_type == "bam"
+            fastq_gz:            it[0].data_type == "fastq_gz"
+            one_fastq_gz_spring: it[0].data_type == "one_fastq_gz_spring"
+            two_fastq_gz_spring: it[0].data_type == "two_fastq_gz_spring"
         }
 
+        // Two fastq.gz-files
+        fastq_gz = input_sample_type.fastq_gz.map { meta, files -> addReadgroupToMeta(meta, files) }
+
+        // Just one fastq.gz.spring-file with both R1 and R2
+        fastq_gz_pair_from_spring = SPRING_DECOMPRESS_TO_FQ_PAIR(input_sample_type.one_fastq_gz_spring, false)
+
+        one_fastq_gz_from_spring = fastq_gz_pair_from_spring.fastq.map { meta, files -> addReadgroupToMeta(meta, files) }
+
+        // Two fastq.gz.spring-files - one for R1 and one for R2
+        r1_fastq_gz_from_spring = SPRING_DECOMPRESS_TO_R1_FQ(input_sample_type.two_fastq_gz_spring.map{ meta, files ->
+            [meta, files[0] ]},
+            true // write_one_fastq_gz
+        )
+        r2_fastq_gz_from_spring = SPRING_DECOMPRESS_TO_R2_FQ(input_sample_type.two_fastq_gz_spring.map{ meta, files ->
+            [meta, files[1] ]},
+            true // write_one_fastq_gz
+        )
+
+        two_fastq_gz_from_spring = r1_fastq_gz_from_spring.fastq.join(r2_fastq_gz_from_spring.fastq).map{ meta, fastq_1, fastq_2 -> [meta, [fastq_1, fastq_2]]}
+
+        two_fastq_gz_from_spring = two_fastq_gz_from_spring.map { meta, files -> addReadgroupToMeta(meta, files) }
```

Comment:
not entirely clear to me why this was not kept in the samplesheet subworkflow, but we can refactor this also later

Reply:
Yeah, I just remember we ran into problems with placing that stuff in the samplesheet subworkflow, and so we put it there.